crates/bevy_math/src/primitives/dim3.rs

```diff
@@ -629,6 +629,132 @@ impl Torus {
     }
 }
 
+/// A 3D triangle primitive.
+#[derive(Clone, Copy, Debug, PartialEq)]
+#[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
+pub struct Triangle3d {
+    /// The vertices of the triangle
+    pub vertices: [Vec3; 3],
+}
+
+impl Primitive3d for Triangle3d {}
+
+impl Default for Triangle3d {
+    /// Returns the default [`Triangle3d`] with vertices at the `[-0.5, 0.0, 0.0]`, `[0.5, 0.0, 0.0]`, and `[0.0, 0.5, 0.0]`
+    fn default() -> Self {
+        Self {
+            vertices: [Vec3::ZERO, Vec3::X, Vec3::Y],
+        }
+    }
+}
+
+impl Triangle3d {
+    /// Create a new [`Triangle3d`] from points `a`, `b`, and `c`
+    #[inline(always)]
+    pub fn new(a: Vec3, b: Vec3, c: Vec3) -> Self {
+        Self {
+            vertices: [a, b, c],
+        }
+    }
+
+    /// Get the area of the triangle
+    #[inline(always)]
+    pub fn area(&self) -> f32 {
+        let [a, b, c] = self.vertices;
+        let ab = a.distance(b);
+        let bc = b.distance(c);
+        let ca = c.distance(a);
+
+        let s = (ab + bc + ca) / 2.0;
+        (s * (s - ab) * (s - bc) * (s - ca)).sqrt()
+    }
+
+    /// Get the perimeter of the triangle
+    #[inline(always)]
+    pub fn perimeter(&self) -> f32 {
+        let [a, b, c] = self.vertices;
+        let ab = a.distance(b);
+        let bc = b.distance(c);
+        let ca = c.distance(a);
+        ab + bc + ca
+    }
+
+    /// Get the normal of the triangle in the direction of the right-hand rule, assuming
+    /// the vertices are ordered in a counter-clockwise direction.
+    ///
+    /// The normal is computed as the cross product of the vectors `ab` and `ac`.
+    #[inline(always)]
+    pub fn normal(&self) -> Dir3 {
+        let [a, b, c] = self.vertices;
+        let ab = b - a;
+        let ac = c - a;
+        Dir3::new_unchecked(ab.cross(ac).normalize())
+    }
+
+    /// Reverse the order of the vertices in the triangle
+    #[inline(always)]
+    pub fn reverse(&mut self) {
+        self.vertices.swap(1, 2);
+    }
+
+    /// Get the centroid of the triangle
+    #[inline(always)]
+    pub fn centroid(&self) -> Vec3 {
```

Comment:
Would it be good to have a doc alias for `center`?

Reply:
I'd also make the doc comment something like "Get the geometric center of the triangle" since that might have more value than repeating the term "centroid" without explaining what it is. Maybe the formula `(a + b + c) / 3.0`could also be mentioned?

With uniform density, I believe the centroid is also the barycenter or center of mass, so those could also be doc aliases.

Reply:
Just committed the new changes. Take a look and let me know if it's better.

Reply:
Looks good to me.